datafusion/sqllogictest/test_files/limit.slt

```diff
@@ -400,7 +400,7 @@ AggregateExec: mode=FinalPartitioned, gby=[i@0 as i], aggr=[]
 --CoalesceBatchesExec: target_batch_size=8192
 ----RepartitionExec: partitioning=Hash([i@0], 4), input_partitions=4
 ------AggregateExec: mode=Partial, gby=[i@0 as i], aggr=[]
---------MemoryExec: partitions=4, partition_sizes=[1, 2, 1, 1]
+--------MemoryExec: partitions=4, partition_sizes=[1, 1, 2, 1]
```

Comment:
I dont see how the PR can affect the partition size distribution as the test plan doesn't have window. More likely the test is odd

Reply:
https://github.com/apache/arrow-datafusion/issues/9450